lib/invoker.js

```diff
@@ -24,21 +25,62 @@ const invoke = (userFunction) => {
     };
 };
 
-module.exports = (userFunctionUri, port) => {
-    const userFunction = lookUpFunction(userFunctionUri);
+const registerShutdown = (destroyHook, timeout, server, shutdownSignal) => {
+    return shutdownSignal.then(() => {
+        logger('Graceful shutdown started');
+        return new Promise(((resolve, reject) => {
+            server.tryShutdown(() => {
```

Comment:
the server may have encountered an err during shutdown. Right now we're ignoring all errors, what should we do?

Reply:
At least partially fixed now, as `server.js` exits appropriately and this promise will be rejected if `tryShutdown` throws.
If `tryShutdown` throws, shall the $destroy hook be executed? If so, the $destroy hook may fail as well and possibly shadow the `tryShutdown` error, what should be surfaced? Both errors? The last one?

Reply:
I meant that tryShutdown can emit an error
```suggestion
            server.tryShutdown(err => {
```

Reply:
Replace `new Promise` with `util.promisify` just around `server.tryShutdown` and made it so that the `$destroy` hook is always called even if `tryShutdown` errors.